src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp

```diff
@@ -42,144 +34,129 @@ enum Spells
-                return;
-
-            //ShadowBolt
-            if (ShadowBolt_Timer <= diff)
+            switch (eventId)
             {
-                if (SelectTarget(SELECT_TARGET_RANDOM, 0))
-                    DoCastVictim(SPELL_SHADOWBOLT);
-
-                ShadowBolt_Timer = 10000;
-            } else ShadowBolt_Timer -= diff;
-
-            //Cleave
-            if (Cleave_Timer <= diff)
-            {
-                DoCastVictim(SPELL_CLEAVE);
-                //13 seconds until we should cast this again
-                Cleave_Timer = 7000 + (rand32() % 10000);
-            } else Cleave_Timer -= diff;
-
-            //MortalStrike
-            if (MortalStrike_Timer <= diff)
-            {
-                DoCastVictim(SPELL_MORTALSTRIKE);
-                MortalStrike_Timer = 10000 + (rand32() % 15000);
-            } else MortalStrike_Timer -= diff;
-
-            //RaiseDead
-            //            if (RaiseDead_Timer <= diff)
-            //          {
-            //      DoCast(me, SPELL_RAISEDEAD);
-            //                RaiseDead_Timer = 45000;
-            //            } else RaiseDead_Timer -= diff;
-
-            //SummonSkeletons
-            if (SummonSkeletons_Timer <= diff)
-            {
-                me->SummonCreature(11197, ADD_POS_1, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_2, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_3, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_4, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_5, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_6, TEMPSUMMON_TIMED_DESPAWN, 29000);
-
-                //34 seconds until we should cast this again
-                SummonSkeletons_Timer = 40000;
-            } else SummonSkeletons_Timer -= diff;
-
-            DoMeleeAttackIfReady();
+                case EVENT_SPELL_SHADOWBOLT:
+                    if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
+                        DoCast(target, SPELL_SHADOWBOLT);
+                    events.Repeat(10s);
+                    break;
+                case EVENT_SPELL_CLEAVE:
+                    DoCastVictim(EVENT_SPELL_CLEAVE);
+                    events.Repeat(7s, 17s);
+                    break;
+                case EVENT_SPELL_MORTALSTRIKE:
+                    DoCastVictim(SPELL_MORTALSTRIKE);
+                    events.Repeat(10s, 25s);
+                    break;
+                case EVENT_SUMMON_SKELETON:
+                    if (!_raiseDead)
+                    {
+                        DoCast(SPELL_RAISEDEAD);
+                        _raiseDead = true;
+                        events.Repeat(1s);
+                    }
+                    else
+                    {
+                        for (uint8 i = 0; i < sizeof(PosSkeleton[0]); ++i)
```

Comment:
You could use std::array instead of c array and use its .size method ;p

Reply:
static initializing std::array sucks, in this case `for (Position const& skeletonPos : PosSkeleton)` will do better

Reply:
or that :P